packages/protocol/contracts/layer2/based/anchor/ShastaAnchor.sol

```diff
@@ -62,10 +68,22 @@ abstract contract ShastaAnchor is PacayaAnchor {
         uint256 parentId = block.number - 1;
         _verifyAndUpdatePublicInputHash(parentId);
         _verifyBaseFeeAndUpdateGasExcess(_parentGasUsed, _baseFeeConfig);
-        _syncChainData(_anchorBlockId, _anchorStateRoot);
         _updateParentHashAndTimestamp(parentId);
 
-        signalService.receiveSignals(_signalSlots);
+        if (_anchorBlockId == 0) {
+            // For blocks that are not the last in the batch, _anchorBlockId, _anchorStateRoot, and
+            // _signalSlots must be zero or empty.
+            require(_anchorStateRoot == 0, NonZeroAnchorStateRoot());
+            require(_signalSlots.length == 0, NonEmptySignalSlots());
+        } else {
+            // For the final block in the batch, _anchorStateRoot must be non-zero.
+            require(_anchorStateRoot != 0, ZeroAnchorStateRoot());
+            _syncChainData(_anchorBlockId, _anchorStateRoot);
+        }
+
+        if (_signalSlots.length != 0) {
+            signalService.receiveSignals(_signalSlots);
+        }
 
         // We need to add one SSTORE from non-zero to non-zero (5000), one addition (3), and one
```

Comment:
Not related to this PR, shall we create an `init2` method in  `TaikoAnchor` to add sth like `lastAnchorGasUsed = 1`, to ensure the `lastAnchorGasUsed` below is non-zero from the beginning?

Reply:
No need, as this feature is not expected to be used right after the fork.